prober/http.go

```diff
@@ -647,7 +647,7 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 			level.Error(logger).Log("msg", "Final request was over SSL")
 			success = false
 		}
-	} else if httpConfig.FailIfNotSSL {
+	} else if httpConfig.FailIfNotSSL && success {
```

Comment:
can we add a test case for it please :)

Reply:
@electron0zero can you check again please?

Reply:
lgtm, approved.